accounts/abi/bind/backends/simulated_test.go

```diff
@@ -1336,3 +1339,129 @@ func TestForkResendTx(t *testing.T) {
+	gasPrice := new(big.Int).Add(head.BaseFee, big.NewInt(1))
+	sim.SetMarketGasPrice(int64(gasPrice.Uint64()))
+
+	lowGasFeeTx := types.NewTx(&types.LegacyTx{
+		Nonce:    uint64(0),
+		To:       &testAddr,
+		Value:    big.NewInt(1),
+		GasPrice: new(big.Int).Sub(gasPrice, big.NewInt(1)),
+		Gas:      gas,
+	})
+	signedTx, err := types.SignTx(lowGasFeeTx, types.HomesteadSigner{}, testKey)
+	require.NoError(t, err)
+
+	// send tx to simulated backend
+	err = sim.SendTransaction(bgCtx, signedTx)
+	require.NoError(t, err)
+	sim.Commit()
+
+	// expect nonce be the same because low gas fee tx will not be mined
+	nonce, err := sim.PendingNonceAt(bgCtx, testAddr)
+	require.NoError(t, err)
+	assert.Equal(t, uint64(0), nonce)
+
+	// send tx with higher gas fee
+	sufficientGasFeeTx := types.NewTx(&types.LegacyTx{
+		Nonce:    uint64(0),
+		To:       &testAddr,
+		Value:    big.NewInt(1),
+		GasPrice: gasPrice,
+		Gas:      gas,
+	})
+
+	signedSufficientTx, err := types.SignTx(sufficientGasFeeTx, types.HomesteadSigner{}, testKey)
+	require.NoError(t, err)
+
+	err = sim.SendTransaction(bgCtx, signedSufficientTx)
+	require.NoError(t, err)
+	sim.Commit()
+
+	// the higher gas transaction should have been mined
+	_, isPending, err := sim.TransactionByHash(bgCtx, signedSufficientTx.Hash())
+	require.NoError(t, err)
+	assert.False(t, isPending)
+
+	// expect nonce has increased
+	nonce, err = sim.PendingNonceAt(bgCtx, testAddr)
+	require.NoError(t, err)
+	assert.Equal(t, uint64(1), nonce)
+}
+
+// TestLowGasTxGetMinedOnceGasFeeDropped checks that lower gas fee txes still stay in mem pool
+// and get mined once gas fee requirement has been met
+// Steps:
+// 1. Send a tx lower than the set market gas price
+// 2. Commit to chain, tx does not get mined
+// 3. Gas fee drops, commit again
+// 4. Check tx get mined
+func TestLowGasTxGetMinedOnceGasFeeDropped(t *testing.T) {
+	testAddr := crypto.PubkeyToAddress(testKey.PublicKey)
+	var gas uint64 = 21000
+
+	sim := simTestBackend(testAddr)
+	defer sim.Close()
+
+	bgCtx := context.Background()
+
+	// Create tx and send
+	head, _ := sim.HeaderByNumber(context.Background(), nil) // Should be child's, good enough
+	normalGasPrice := new(big.Int).Add(head.BaseFee, big.NewInt(1))
+	highGasPrice := new(big.Int).Add(head.BaseFee, big.NewInt(5))
+
+	sim.SetMarketGasPrice(int64(highGasPrice.Uint64()))
+
+	normalGasFeeTx := types.NewTx(&types.LegacyTx{
+		Nonce:    uint64(0),
+		To:       &testAddr,
+		Value:    big.NewInt(1),
+		GasPrice: normalGasPrice,
+		Gas:      gas,
+	})
+	signedTx, err := types.SignTx(normalGasFeeTx, types.HomesteadSigner{}, testKey)
+	require.NoError(t, err)
+
+	// send tx to simulated backend
+	err = sim.SendTransaction(bgCtx, signedTx)
+	require.NoError(t, err)
+	sim.Commit()
+
+	// check that the nonce stays the same because nothing has been mined
+	pendingNonce, err := sim.PendingNonceAt(bgCtx, testAddr)
+	require.NoError(t, err)
+	assert.Equal(t, uint64(0), pendingNonce)
```

Comment:
let's do another commit and nonce check after this one? so we go through the cycle of passing forward the stuckTransactions twice.

Reply:
this test set up is the following
(1). Send a tx lower than the set market gas price >> tx ends up in stuck transactions
(2). call commit to chain >> tx stills in stuck transactions
(3). drop gas fee, call commit to chain >> stuck transactions txes get mined.

hmmm, are you suggesting after (3) we should commit again even though we dont have new tx and nothing in stuck transactions?